test/libyul/yulOptimizerTests/fullSuite/loopInvariantCodeMotion.yul

```diff
@@ -0,0 +1,32 @@
+{
+    sstore(0, array_sum(calldataload(0)))
+
+    function array_sum(x) -> sum {
+        let length := calldataload(x)
+        for { let i := 0 } lt(i, length) { i := add(i, 1) } {
+            sum := add(sum, array_load(x, i))
+        }
+    }
+    function array_load(x, i) -> v {
+        let len := calldataload(x)
+        if iszero(lt(i, len)) { revert(0, 0) }
+        let data := add(x, 0x20)
+        v := calldataload(add(data, mul(i, 0x20)))
+        v := add(v, calldataload(7))
```

Comment:
Is this supposed to be `array_load_and_add_constant`? Not sure I follow the meaning of `calldataload(7)`

Reply:
It is non-sensical, I just wanted to add some code.